src/server/server_family.cc

```diff
@@ -526,6 +598,17 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
   stats_caching_task_ =
       pb_task_->AwaitBrief([&] { return pb_task_->AddPeriodic(period_ms, cache_cb); });
 
+  // check for '--replicaof' before loading anything
+  if (ReplicaOfFlag flag = GetFlag(FLAGS_replicaof); flag) {
+    bool success = false;
+    service_.proactor_pool()[0].Await([this, flag = move(flag), &success]() {
+      this->Replicate(move(flag.host), move(flag.port), &success);
+    });
+
+    if (!success && !GetFlag(FLAGS_continue_on_replication_fail))
+      exit(EXIT_FAILURE);
```

Comment:
I'd add a `LOG(ERROR)` here just to be 100% sure that users understand why the process exited.